### ISSUE TITLE
[FR] Enable Toggle Option for Entity

### ISSUE THREAD

**Issue by shbatm** [rianadon/timer-bar-card#43]
First thanks for the card!

This is usable for the RainMachine integration with:

```yaml
# On Zones:
duration:
  units: seconds
  attribute: time_remaining
# On Programs:
start_time:
  attribute: next_run
```

However, each zone and program is also a switch, and I would like to see an option to keep the toggle on the far right of the entity row. Basically combine this into 1 row:

![image](https://user-images.githubusercontent.com/18057952/166257903-fabdd2f9-1299-4181-8464-2df345b9f47f.png)

Also, it would be nice to be able to select the timer status attribute separately from the state. 

For example, RainMachine with show all of the zone states as "ON" when a program is active, but it has some modes that will actually cycle between zones to let things soak in. Therefore you can only tell if a zone is actually flowing by using the `status` attribute which uses "running" for `active_states` and `pending` if "on-but-paused" or `not running` when the zone/program is off.

Program:
![image](https://user-images.githubusercontent.com/18057952/166256526-d0073a91-e728-440d-a6d8-7a0959b7839d.png)

Zone:
![image](https://user-images.githubusercontent.com/18057952/166256631-7b6a6060-0418-4cc4-b447-0650a1e94bc2.png)

**Comment by rianadon** [comment]
> First thanks for the card!

You're welcome! 😍

> However, each zone and program is also a switch, and I would like to see an option to keep the toggle on the far right of the entity row. Basically combine this into 1 row:

Unfortunately, I don't see a way to implement this without a lot of work and time on my end. Personally, I'd rather keep the card's functionality to only displaying timers and not controlling other entities.

I imagine more complicated use cases could get their own cards, like [my OpenSprinkler card](https://github.com/rianadon/opensprinkler-card) which combines the timer with a button. I don't have the time to write a version for RainMachine but could help anyone trying to do so.

> For example, RainMachine with show all of the zone states as "ON" when a program is active, but it has some modes that will actually cycle between zones to let things soak in. Therefore you can only tell if a zone is actually flowing by using the `status` attribute which uses "running" for `active_states` and `pending` if "on-but-paused" or `not running` when the zone/program is off.

Would adding some sort of ability to determine the state based on attributes would solve this issue?

**Comment by shbatm** [comment]
> Unfortunately, I don't see a way to implement this without a lot of work and time on my end.

Understandable. I tried to combine it with [extend_paper_buttons_row](https://github.com/jcwillox/lovelace-paper-buttons-row#embedding-in-other-entity-rows) which I use elsewhere to tack a button on the end of a row, but it didn't work with this card. I'll have to investigate further, it's probably in the way the timer is rendered.

> Yes, I think the ability to set an attribute other than state for active/idle/paused_state would work. Similar to the logic/config for statt_time & duration.

**Comment by rianadon** [comment]
> Understandable. I tried to combine it with [extend_paper_buttons_row](https://github.com/jcwillox/lovelace-paper-buttons-row#embedding-in-other-entity-rows) which I use elsewhere to tack a button on the end of a row, but it didn't work with this card. I'll have to investigate further, it's probably in the way the timer is rendered.

That's a very amazing card! I'll look into it as well. It would be awesome if they could work together.

> Yes, I think the ability to set an attribute other than state for active/idle/paused_state would work. Similar to the logic/config for statt_time & duration.

Awesome!

**Comment by rianadon** [comment]
These issues are addressed in the newly released v1.18!
